frb_codegen/src/commands.rs

```diff
@@ -254,15 +254,21 @@ pub fn format_dart(path: &str, line_length: i32) -> Result {
 
 pub fn build_runner(dart_root: &str) -> Result {
     info!("Running build_runner at {}", dart_root);
+    // what is the correct way to differentiate a Dart project from a Flutter project ?
+    // or is it better to let the user specify it, let's say with an env var ?
+    let is_dart = true; // FIXME: determine whether it's a Dart or Flutter project
+    let cmd = if is_dart { "dart run build_runner build" } else { "flutter pub run build_runner build" };
```

Comment:
uber-nit: maybe we can extract the common "run build_runner build"?

Reply:
Yes, but there's a slight difference:
- in dart : `dart run build_runner build` (no `pub`)
- in flutter : `flutter pub run build_runner build` (notice the mandatory `pub`)

Reply:
I tested both on my external project :
:white_check_mark: _with pub_

```sh
➜  cerclo-app git:(main) ✗ flutter pub run build_runner build
[INFO] Generating build script...
[INFO] Generating build script completed, took 171ms

[INFO] Initializing inputs
[INFO] Reading cached asset graph...
[INFO] Reading cached asset graph completed, took 25ms

[INFO] Checking for updates since last build...
[INFO] Checking for updates since last build completed, took 279ms

[INFO] Running build...
[INFO] Running build completed, took 5ms

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 16ms

[INFO] Succeeded after 25ms with 0 outputs (0 actions)
```

:x: _without pub_
```sh
➜  cerclo-app git:(main) ✗ flutter run build_runner build 
Target file "build_runner" not found.
```

Reply:
Sure. So maybe "flutter pub" and "dart", i.e. `build_runner build` are indeed the same

Reply:
ok what about 59e00c119ef3bba900efef2fd2e5940b40da19bc ?

Reply:
sorry this one too 86dba0970f1daf016a37f95a583ed94b8bda8671

Reply:
do you want to simplify even some more while we're at it ?
<img width="602" alt="Screen Shot 2565-07-29 at 20 16 32" src="https://user-images.githubusercontent.com/21016014/181768611-cdbc96fa-8408-40c5-a38c-f623a1758b9e.png">

Reply:
LGTM :)

Reply:
Ok if no further modifications, let's resolve this one.